### PR TITLE
fix: handling of active_count

### PR DESCRIPTION
**Pull request by abc3**

### CODE REVIEW

**Comment by chasers** [comment]
How do we have a test for this?

**Comment by abc3** [comment]
for example, add a test for session and transaction modes where the transaction contains 100+ queries

File: lib/supavisor/db_handler.ex [hunk 1]
```diff
@@ -297,8 +297,10 @@ defmodule Supavisor.DbHandler do
     if String.ends_with?(bin, Server.ready_for_query()) do
       {_, stats} = Telem.network_usage(:db, data.sock, data.id, data.stats)
 
+      # in transaction mode, we need to notify the client when the transaction is finished,
+      # after which it will unlink the direct db connection process from itself.
       data =
-        if data.mode in [:transaction, :session] do
+        if data.mode == :transaction do
```
Comment (hauleth) [inline lib/supavisor/db_handler.ex:303]: Wasn't that change added recently? It is reverting it?

**Review by hauleth — APPROVED** [review]

File: lib/supavisor/db_handler.ex (hunk 1 above)
Comment (abc3) [reply lib/supavisor/db_handler.ex:303]: Yes, but we don't need it. The ready_for_query is used to indicate that the client process needs to be unlinked from the direct connection. In session mode, it is always linked

**Pull request merged**